docs/specifications/tests/Methods.md

```diff
@@ -1,69 +1,280 @@
-   from Method 2. 
-2. Record all the unique name servers from the answers received from the query in 
-   step 1.
+* Unless *undelegated test*, the name servers to the parent zone as found 
+  in [BASIC01] ("parent NS").
+* The name of the child zone ("child zone").
+* The fact if the child zone exists from [BASIC01].
+* The fact if the test type is undelegated test ("undelegate test") or not
+  ("normal test").
+* If *undelegated test*, the submitted name server data.
 
-### Method 4: Obtain "glue address records" from parent
+### Ordered description of steps to be taken to execute the method
 
-This method tries to obtain any glue address records from the delegation
-in the parent zone.
+1. *Normal test*. For *undelegated test* go to 2.
 
-1. Query the servers in Method 2 for A and AAAA addresses.
-2. Record the unique IP addresses from the answers (both A and AAAA) in
-   the additional section, which are the glue address records for the
-   domain.
+   1. If child zone does not exists then return ERROR.
+   2. These steps assume that the servers of the parent zone behaves the
+      same way as when [BASIC01] was run.
+   3. Create an SOA query for the child zone and send that one server of
+      *parent NS* with RD flag unset.
+   4. If the response response contains a referal to the child zone:
+      1. Extract the name server names from the RDATA of the NS records in
+         the authority section.
+      2. Return the set of name servers (name server names).
+      3. Processing the steps is stopped.
+   5. If the response is authoritative (AA bit set) and the answer section
+      contains the SOA record of the child zone:
+      1. Repeat the query with the next server of *parent NS* until a 
+         server has responded with a referal or no more servers are available.
+      2. If a referal was found, go to step 4.
+      3. If no referal was found, send a query for the NS records instead,
+         and use them as if it was a referal.
+      4. Return the set of name servers (name server names).
+      5. Processing the steps is stopped.
 
-### Method 5: Obtain the name server address records from child
+2. *Undelegated test*.
 
-This method tries to obtain the IP addresses for the name servers used in
-the child zone.
+   1. Collect the name server names provided as indata for the test.
+   2. Return the set of name servers (name server names).
+   3. Processing the steps is stopped.
 
-1. Send an A query to all name servers obtained in Method 3.
-2. Record the list of unique IPv4 addreses in the answer section.
-3. Send an AAAA query to all name servers obtained in Method 3.
-4. Record the list of unique IPv6 addresses in the answer section.
 
--------
+### Outcome(s)
 
-Copyright (c) 2013, 2014, 2015, IIS (The Internet Infrastructure Foundation)  
-Copyright (c) 2013, 2014, 2015, AFNIC  
-Creative Commons Attribution 4.0 International License
+Unless ERROR was returned in the steps, outcome is the set of name 
+servers.
```

Comment:
"the set of" -> "the non-empty set of". Knowing that is very helpful for the caller.

Reply:
Updated.